qcodes/tests/dataset/test_database_extract_runs.py

```diff
@@ -0,0 +1,477 @@
+    extract_runs_into_db(source_path, target_path, *exp_1_run_ids[1:3])
+
+    assert len(target_exp1) == 3
+
+    # insert run from different experiment
+    extract_runs_into_db(source_path, target_path, ds.run_id)
+
+    assert len(target_exp1) == 3
+
+    target_exp2 = Experiment(conn=target_conn, exp_id=2)
+
+    assert len(target_exp2) == 1
+
+    # finally insert every single run from experiment 1
+
+    extract_runs_into_db(source_path, target_path, *exp_1_run_ids)
+
+    # check for idempotency once more by inserting all the runs but in another
+    # order
+    with raise_if_file_changed(target_path):
+        extract_runs_into_db(source_path, target_path, *exp_1_run_ids[::-1])
+
+    target_exps = get_experiments(target_conn)
+
+    assert len(target_exps) == 2
+    assert len(target_exp1) == 5
+    assert len(target_exp2) == 1
+
+    # check that all the datasets match up
+    for run_id in exp_1_run_ids + exp_2_run_ids:
+        source_ds = DataSet(conn=source_conn, run_id=run_id)
+        target_ds = load_by_guid(guid=source_ds.guid, conn=target_conn)
+
+        assert source_ds.the_same_dataset_as(target_ds)
+
+        source_data = source_ds.get_data(*source_ds.parameters.split(','))
+        target_data = target_ds.get_data(*target_ds.parameters.split(','))
+
+        assert source_data == target_data
+
+
+def test_runs_from_different_experiments_raises(two_empty_temp_db_connections,
+                                                some_paramspecs):
+    """
+    Test that inserting runs from multiple experiments raises
+    """
+    source_conn, target_conn = two_empty_temp_db_connections
+
+    source_path = path_to_dbfile(source_conn)
+    target_path = path_to_dbfile(target_conn)
+
+    source_exp_1 = Experiment(conn=source_conn)
+    source_exp_2 = Experiment(conn=source_conn)
+
+    # make 5 runs in first experiment
+
+    exp_1_run_ids = []
+    for _ in range(5):
+
+        source_dataset = DataSet(conn=source_conn, exp_id=source_exp_1.exp_id)
+        exp_1_run_ids.append(source_dataset.run_id)
+
+        for ps in some_paramspecs[2].values():
+            source_dataset.add_parameter(ps)
+
+        for val in range(10):
+            source_dataset.add_result({ps.name: val
+                                       for ps in some_paramspecs[2].values()})
+        source_dataset.mark_complete()
+
+    # make 5 runs in second experiment
+
+    exp_2_run_ids = []
+    for _ in range(5):
+
+        source_dataset = DataSet(conn=source_conn, exp_id=source_exp_2.exp_id)
+        exp_2_run_ids.append(source_dataset.run_id)
+
+        for ps in some_paramspecs[2].values():
+            source_dataset.add_parameter(ps)
+
+        for val in range(10):
+            source_dataset.add_result({ps.name: val
+                                       for ps in some_paramspecs[2].values()})
+        source_dataset.mark_complete()
+
+    run_ids = exp_1_run_ids + exp_2_run_ids
+    source_exp_ids = np.unique([1, 2])
```

Comment:
It should've been this, right?
```suggestion
    source_exp_ids = [source_exp_1.exp_id, source_exp_2.exp_id]
```

Reply:
It should match the match.